kernel/fs/iso9660.inc

```diff
@@ -736,3 +736,1044 @@ cd_compare_name:
+
+;-----------------------------------------------------------------------------
+; ISO9660 external functions
+;   in:
+; ebx -> parameter structure of sysfunc 70
+; ebp -> ISO9660 structure
+; esi -> path string in UTF-8
+;   out:
+; eax, ebx = return values for sysfunc 70
+iso9660_GetFileInfo:
+        cmp     byte[esi], 0
+        jz      .rootdir
+
+        sub     esp, 4 ; for ptr on memory page
+        call    iso9660_find_file
+
+        mov     ecx, [ebx + 16] ; buffer
+
+        call    .copy_file_info
+
+        call    kernel_free
+        xor     eax, eax
+        mov     ebx, 40
+        ret
+
+;  IN: eax -> ISO966_DIRECTORY_RECORD
+;      ecx -> buffer
+; destruct: edx
+.copy_file_info:
+         ; copy size
+        mov     [ecx + 36], dword 0
+        mov     edx, [eax + ISO9660_DIRECTORY_RECORD.data_length]
+        mov     [ecx + 32], edx
+
+        ; copy flags(dir of file)
+        xor     edx, edx
+        or      dl, 000001b
+        test    byte[eax + ISO9660_DIRECTORY_RECORD.flags], 1b ; check hidden flag
+        jz      @f
+        or      dl, 10b
+@@:
+        test    byte[eax + ISO9660_DIRECTORY_RECORD.flags], 10b ; check dir
+        jz      @f
+        or      dl, 10000b ; dir flag
+        mov     dword[ecx + 32], 0 ; size = zero
+@@:
+        mov     [ecx], edx
+
+        ; copy date creat file
+        movzx   edx, byte[eax + ISO9660_DIRECTORY_RECORD.date_time]
+        add     edx, 1900 ; year
+        shl     edx, 8
+
+        mov     dl, byte[eax + ISO9660_DIRECTORY_RECORD.date_time + 1] ;month
+        shl     edx, 8
+        mov     dl, byte[eax + ISO9660_DIRECTORY_RECORD.date_time + 2] ;day
+
+        mov     [ecx + 12], edx
+        mov     [ecx + 20], edx
+        mov     [ecx + 28], edx
+
+        ; copy time creat file
+        movzx   edx, byte[eax + ISO9660_DIRECTORY_RECORD.date_time + 3] ;hour
+        shl     edx, 8
+        mov     dl, byte[eax + ISO9660_DIRECTORY_RECORD.date_time + 4] ; minute
+        shl     edx, 8
+        mov     dl, byte[eax + ISO9660_DIRECTORY_RECORD.date_time + 5] ; second
+
+        mov     [ecx + 8], edx
+        mov     [ecx + 16], edx
+        mov     [ecx + 24], edx
+
+        ret
+
+.rootdir:
+        mov     edi, [ebx + 16] ; edi = buffer
+        ; copy flags (dir)
+        mov     byte [edi], 8
+        ; copy size drive
+        mov     eax, dword[ebp + PARTITION.Length+DQ.lo]
+        mov     edx, dword[ebp + PARTITION.Length+DQ.hi]
+        mov     ecx, [ebp + PARTITION.Disk]
+        mov     ecx, [ecx + DISK.MediaInfo.SectorSize]
+        bsf     ecx, ecx
+        shld    edx, eax, cl
+        shl     eax, cl
+        mov     [edi + 32], eax  ; bdfe.size.lo
+        mov     [edi + 36], edx  ; bdfe.size.hi
+
+        mov     eax, [ebx + 8]
+        ; copy encoding
+        mov     [edi + 4], eax
+        ; check encoding on fs struct
+        test    eax, eax ; check f70s5arg.xflags
+        jz      .no_name
+
+        ; stdcall is_region_userspace, edi, ecx
```

Comment:
why commented out?

Reply:
this check should be done by the kernel, not the driver

Reply:
> this check should be done by the kernel, not the driver

How is the reader of the code supposed to guess what you just said after seeing the commented out line? Maybe you should write this information in comment